docs/configuration.md

```diff
@@ -0,0 +1,772 @@
+---
+foo(arg1, arg2)
+
+function(arg1, // align by (
+    arg2,
+ arg3)
+
+function(
+  argument1,
+  argument2)
+```
+
+```scala mdoc:scalafmt
+align.openParenCallSite = false
+---
+foo(arg1, arg2)
+
+function(arg1, // no align by (
+    arg2,
+ arg3)
+
+function(
+  argument1,
+  argument2)
+```
+
+### `align.openParenDefnSite`
+
+```scala mdoc:defaults
+align.openParenDefnSite
+```
+
+```scala mdoc:scalafmt
+align.openParenDefnSite = true
+---
+class IntString(int: Int, string: String)
+
+class IntStringLong(int: Int,
+    string: String,
+  long: Long)
+```
+
+```scala mdoc:scalafmt
+align.openParenDefnSite = false
+---
+class IntString(int: Int, string: String)
+
+class IntStringLong(
+      int: Int,
+    string: String,
+  long: Long
+)
+```
+
+## Newlines
+
+The `newlines.*` options are used to configure when and where `scalafmt` should
+insert newlines.
+
+> You might be interested in the [Vertical Multiline](#vertical-multiline) section.
+
+### `newlines.alwaysBeforeTopLevelStatements`
+
+```scala mdoc:defaults
+newlines.alwaysBeforeTopLevelStatements
+```
+
+```scala mdoc:scalafmt
+newlines.alwaysBeforeTopLevelStatements = false
+---
+import org.scalafmt
+package core { // no newline added here
+  object O { // nor here
+    val x1 = 1
+    val x2 = 2
+    def A = "A"
+    def B = "B"
+  }
+}
+```
+
+```scala mdoc:scalafmt
+newlines.alwaysBeforeTopLevelStatements = true
+---
+import org.scalafmt
+package core {
+  object O {
+    val x1 = 1
+    val x2 = 2
+    def A = "A"
+    def B = "B"
+  }
+}
+```
+
+### `newlines.alwaysBeforeElseAfterCurlyIf`
+
+```scala mdoc:defaults
+newlines.alwaysBeforeElseAfterCurlyIf
+```
+
+```scala mdoc:scalafmt
+newlines.alwaysBeforeElseAfterCurlyIf = true
+---
+if (someCond) {
+  foo()
+} else {
+  bar()
+}
+```
+
+```scala mdoc:scalafmt
+newlines.alwaysBeforeElseAfterCurlyIf = false
+---
+if (someCond) {
+  foo()
+}
+else {
+  bar()
+}
+```
+
+## Rewrite Rules
+
+To enable a rewrite rule, add it to the config like this
+`rewrite.rules = [SortImports]`.
+
+### `AvoidInfix`
+
+```scala mdoc:scalafmt
+rewrite.rules = [AvoidInfix]
+---
+a success b
+a error (b, c)
+a map { x =>
+  x + 2
+}
+"o" % "a" % "v" c(D)
+future recover {
+  case e: Err => 0
+} map (_.toString)
+```
+
+### `ExpandImportSelectors`
+
+```scala mdoc:scalafmt
+rewrite.rules = [ExpandImportSelectors]
+---
+import a.{
+    b,
+    c
+  }, h.{
+    k, l
+  }
+import d.e.{f, g}
+import a.{
+    foo => bar,
+    zzzz => _,
+    _
+  }
+```
+
+### `RedundantBraces`
+
+> Warning. This rewrite can cause non-idempotent formatting, see
+> [#1055](https://github.com/scalameta/scalafmt/issues/1055).
+
+```scala mdoc:scalafmt
+rewrite.rules = [RedundantBraces]
+---
+def foo = {
+  List(1, 2, 3).sum
+}
+```
+
+```scala mdoc:scalafmt
+rewrite.rules = [RedundantBraces]
+rewrite.redundantBraces.stringInterpolation = true
+---
+q"Hello ${name}"
+```
+
+Configuration options and default values:
+
+// TODO(olafur): multiline defaults
```

Comment:
If you're bored this looks like an easy fix

Reply:
I'll take a look 👍